application/app/src/main/java/com/example/csi_dmce/attendance/Attendance.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.csi_dmce.attendance
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.csi_dmce.database.*
+import kotlinx.coroutines.runBlocking
+
+class Attendance: AppCompatActivity() {
+    var click_count: Int = 0
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        runBlocking {
+            val studentID:String = intent.getStringExtra("student_ID").toString()
+            //where to get the eventID from?
+            // val eventID : String = intent.getStringExtra("event_id").toString()
+
+            val eventId: String = "DCF-1004207400"
+            val eventUUID: String = run { EventWrapper.getEventUUID(eventId)!!}
+
+            //when scan button is clicked (also increment click_count first)
+            val intent = Intent(applicationContext, scan_qr::class.java)
+            startActivity(intent)
+
+            val QRContents:String = intent.getStringExtra("QR_contents").toString()
+
+            if(QRContents!=null){
+                //compare these contents to the ones from the QR generated by admin class
+                when(click_count){
+                    1-> AttendanceWrapper.setQRAttendance(studentID, eventUUID, "first", true)
+                    2-> AttendanceWrapper.setQRAttendance(studentID, eventUUID, "second", true)
+                    else -> return@runBlocking
+                }
+            }
+            val attStatus1 = AttendanceWrapper.checkAttendanceStatus(studentID, eventUUID, "first")
+            val attStatus2 = AttendanceWrapper.checkAttendanceStatus(studentID, eventUUID, "second")
```

Comment:
Short circuit to,

```kotlin
val attendanceObject = AttendanceWrapper.getAttendanceDocument()
val first: Boolean = attendanceObject.first
val second: Boolean = attendanceObject.second 
```

to avoid 2 DB transactions.